File: matrix.py
import numpy, sys, time

def calculate(n):
    a = numpy.zeros((n, n)) # Matrix A
    b = numpy.zeros((n, n)) # Matrix B
    c = numpy.zeros((n, n)) # Matrix C

    # Initialize the matrices to some values.
    for i in range(n):
        for j in range(n):
            a[i, j] = i * n + j
            b[i, j] = j * n + i
            c[i, j] = 0

    begin = time.time()

    # Write code to calculate C = A * B. #
    for i in range(n):
        for k in range(n):
            for j in range(n):
                c[i, k] += a[i, j] * b[j, k]

    end = time.time()
#     print("time: %.6f sec" % (end - begin)) 
    print("%d %.6f" % (n, (end - begin)))
    # Print C for debugging. Comment out the print before measuring the execution time.
    total = 0
    for i in range(n):
        for j in range(n):
            # print c[i, j]
            total += c[i, j]
    # Print out the sum of all values in C.
    # This should be 450 for N=3, 3680 for N=4, and 18250 for N=5.
#     print("sum: %.6f" % total)

# if (len(sys.argv) != 2):
#     print("usage: python %s N" % sys.argv[0])
#     quit()
for i in range(500):
    calculate(i)

File: matrix.java
class Matrix {
  public static void main(String args[])
    {
      // if (args.length != 1) {
      //   System.out.println("usage: java Matrix N");
      //   return;
      // }
      // int n = Integer.parseInt(args[0]);
      int n, m;
      double sum;
      for(n = 2; n < 300;){
        sum = 0;
        for(m = 0; m < 50; m++){
          sum += calculate(n);
        }
        System.out.printf("%d %.6f\n", n, sum/50);
        n += 10;
      }
    }
  
  public static double calculate(int n) {
    double[][] a = new double[n][n]; // Matrix A
    double[][] b = new double[n][n]; // Matrix B
    double[][] c = new double[n][n]; // Matrix C

    // Initialize the matrices to some values.
    int i, j, k;
    for (i = 0; i < n; i++) {
      for (j = 0; j < n; j++) {
        a[i][j] = i * n + j;
        b[i][j] = j * n + i;
        c[i][j] = 0;
      }
    }

    long begin = System.currentTimeMillis();

    /* Write code to calculate C = A * B. */
    for (i = 0; i < n; i++) {
      for (k = 0; k < n; k++) {
        for (j = 0; j < n; j++) {
          c[i][k] += a[i][j] * b[j][k];
        }
      }
    }

    long end = System.currentTimeMillis();
    // System.out.printf("%d %.6f\n", n, (end - begin) / 1000.0);
    // System.out.printf("time: %.6f sec\n", (end - begin) / 1000.0);

    // Print C for debugging. Comment out the print before measuring the execution time.
    double sum = 0;
    for (i = 0; i < n; i++) {
      for (j = 0; j < n; j++) {
        sum += c[i][j];
        // System.out.printf("c[%d][%d]=%f\n", i, j, c[i][j]);
      }
    }
    // Print out the sum of all values in C.
    // This should be 450 for N=3, 3680 for N=4, and 18250 for N=5.
    // System.out.printf("sum: %.6f\n", sum);
    return (end - begin) / 1000.0;
  }
}
